smart_open/smart_open_lib.py

```diff
@@ -382,7 +382,7 @@ def open(
 
 def smart_open(uri, mode="rb", **kw):
     """Deprecated, use smart_open.open instead."""
-    logger.warning('this function is deprecated, use smart_open.open instead')
+    warnings.warn('this function is deprecated, use smart_open.open instead')
```

Comment:
Can we link to the `smart_open` => `open` migration instructions here? This "use smart_open.open instead" is a massive ask, unless it's a 1:1 swap without any potential issues (which I don't think it is).

Let's inform the user more clearly what are the expected tradeoffs / benefits of such change, and what their migration and potential incompatibilities look like.